### PR TITLE
add new semantics on resource group and removing memory model.

### PR DESCRIPTION
**Pull request by SmartKeyerror**
After #14343, it's time to remove all the relevant codes and test cases about the 
resource group memory manager. 

### 1. What this PR has done

This PR did most 2 important things:

1. First, the most important is to remove all the codes and test cases about resource
group memory model, which includes the functions, variables, GUCs, etc.
2. add new semantics on resource group and removing memory model.

Since `pg_resgroupcapability.reslimittype` is consistent with the enumerated type 
`ResGroupLimitType`, when we delete `RESGROUP_LIMIT_TYPE_MEMORY and other content, 
there will be "holes". In order to avoid more PR and review work, this PR deleted
memory model and added new semantics.

The GUC this PR removed:

`gp_resource_group_memory_limit`, `gp_resgroup_memory_policy`,
`memory_spill_ratio`, `gp_log_resgroup_memory`, `gp_resgroup_memory_policy_auto_fixed_mem`
`gp_resource_group_cpu_ceiling_enforcement`, `gp_resgroup_print_operator_memory_limits`
`gp_resource_group_enable_recalculate_query_mem`.

### 2. New Resource Group Attributes and Limits

New Resource group attributes and limits:

1. `concurrenty`. The maximum number of concurrent transactions, including active and 
idle transactions, that are permitted in the resource group.
2. `cpu_hard_quoata_limit`. The percentage of CPU resources hard limit to this resource 
group. This value indicates the maximum CPU ratio that the current group can use.
3. `cpu_soft_priority`. The current group CPU priority, the larger the value, the higher the 
priority, the more likely to be scheduled by the CPU, the default value is 100.
4. `cpuset`. The CPU cores to reserve for this resource group.

First, let's take a look at the new resource management view of resource group:

```plain
postgres=# select * from gp_toolkit.gp_resgroup_config;
 groupid |   groupname   | concurrency | cpu_hard_quota_limit | cpu_soft_priority | cpuset 
---------+---------------+-------------+----------------------+-------------------+--------
    6437 | default_group | 20          | 20                   | 100               | -1
    6438 | admin_group   | 10          | -1                   | 300               | -1
    6441 | system_group  | 0           | 10                   | 100               | -1
(3 rows)
```

#### 2.1 What's the meaning of `cpu_hard_quota_limit`

It can be seen that `cpu_rate_limit` is removed and replaced by `cpu_hard_quota_limit`, which 
indicates the upper limit of CPU resources that the current group can use. This is a percentage, 
taking 20 as an example, it means that the CPU resources used by the current group cannot exceed 
20% of the total CPU resources of the Host.

The sum of `cpu_hard_quota_limit` of all groups can exceed 100, and the range of this value is 
[1, 100] or -1, where 100 and -1 both mean that all CPU resources can be used, and no CPU resource 
limit is imposed on it.

When we change the value of `cpu_hard_quota_limit`, will write

```
cpu.cfs_period_us * ncores * cpu_hard_quota_limit / 100
```

to the file `cpu.cfs_quota_us`.

#### 2.2 What's the meaning of `cpu_soft_priority`

We have added `cpu_soft_priority` this field, which is used to indicate the CPU priority of the current 
group, corresponding to the dynamic running load weight in Linux CFS. The larger the value, the greater 
the weight of the group, and it will be scheduled more preferentially by the Linux scheduling process.

The value range is [1, +∞], currently, the value cannot exceed 2^64 - 1. The default value is 100.

When we change the value of `cpu_soft_priority`, will write

```
(int64)(cpu_soft_priority * 1024 / 100)
```

to the file `cpu.shares`.

### CODE REVIEW

File: src/test/isolation2/isolation2_resgroup_schedule [hunk 1]
```diff
@@ -11,38 +11,19 @@ test: resgroup/resgroup_name_convention
...
 # functions
 test: resgroup/resgroup_concurrency
 test: resgroup/resgroup_bypass
-test: resgroup/resgroup_bypass_memory_limit
-test: resgroup/resgroup_alter_concurrency
-test: resgroup/resgroup_memory_statistic
-test: resgroup/resgroup_memory_limit
-test: resgroup/resgroup_memory_runaway
-test: resgroup/resgroup_alter_memory
+#test: resgroup/resgroup_alter_concurrency
```
Comment (RMTT) [inline src/test/isolation2/isolation2_resgroup_schedule:18]: I just think this test case can be opened now.

File: src/test/isolation2/sql/resgroup/resgroup_parallel_queries.sql [hunk 2]
```diff
@@ -78,12 +78,9 @@ LANGUAGE 'plpgsql';
...
-6>:select exec_commands_n('dblink_rg_test6','CREATE RESOURCE GROUP rg_test_g# WITH (concurrency=#, cpu_rate_limit=#, memory_limit=#)', 'DROP RESOURCE GROUP rg_test_g#', 'ALTER RESOURCE GROUP rg_test_g# set memory_limit #', 60, '', '1-6', false);
+1>:select exec_commands_n('dblink_rg_test1','CREATE RESOURCE GROUP rg_test_g# WITH (concurrency=#, cpu_hard_quota_limit=#)', 'DROP RESOURCE GROUP rg_test_g#', 'ALTER RESOURCE GROUP rg_test_g# set concurrency #', 60, '', '1-6', false);
+2>:select exec_commands_n('dblink_rg_test2','CREATE RESOURCE GROUP rg_test_g# WITH (concurrency=#, cpu_hard_quota_limit=#)', 'DROP RESOURCE GROUP rg_test_g#', 'ALTER RESOURCE GROUP rg_test_g# set concurrency#', 60, '', '1-6', false);
+3>:select exec_commands_n('dblink_rg_test3','CREATE RESOURCE GROUP rg_test_g# WITH (concurrency=#, cpu_hard_quota_limit=#)', 'DROP RESOURCE GROUP rg_test_g#', 'ALTER RESOURCE GROUP rg_test_g# set cpu_hard_quota_limit #', 60, '', '1-6', false);
```
Comment (RMTT) [inline src/test/isolation2/sql/resgroup/resgroup_parallel_queries.sql:83]: Why reduce the cases here.

**Review by RMTT — COMMENTED** [review]
Nice working!

File: src/test/isolation2/sql/resgroup/resgroup_parallel_queries.sql (hunk 2 above)
Comment (SmartKeyerror) [reply src/test/isolation2/sql/resgroup/resgroup_parallel_queries.sql:83]: Since we don't need to test the memory, just remove it.

**Comment by SmartKeyerror** [comment]
My local full pipeline test cases are all passed, so merge this PR.

**Pull request merged**